components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/behavior/BrowserGestureDetector.kt

```diff
@@ -29,14 +29,19 @@ internal class BrowserGestureDetector(
     @Suppress("MaxLineLength")
     internal var gestureDetector = GestureDetector(
         applicationContext,
-        CustomScrollDetectorListener { previousEvent: MotionEvent, currentEvent: MotionEvent, distanceX, distanceY ->
+        CustomScrollDetectorListener { previousEvent: MotionEvent?, currentEvent: MotionEvent, distanceX, distanceY ->
             run {
                 listener.onScroll?.invoke(distanceX, distanceY)
 
-                if (abs(currentEvent.y - previousEvent.y) >= abs(currentEvent.x - previousEvent.x)) {
-                    listener.onVerticalScroll?.invoke(distanceY)
-                } else {
-                    listener.onHorizontalScroll?.invoke(distanceX)
+                // We got many crashes because of the initial event - ACTION_DOWN being null.
+                // Investigations to be continued in android-components/issues/8552.
+                // In the meantime we'll protect against this with a simple null check.
+                if (previousEvent != null) {
```

Comment:
Should we add a test as part of the change?

Reply:
Added one that would fail with the same stacktrace before the patch.